server/app.js

```diff
@@ -1,24 +1,85 @@
 'use strict';
 
+var async = require('async');
 var db = require('../database.js');
 var parseService = require('./services/parsing/readFilesForParsing.js');
 var downloadService = require('./services/downloading/gitHubRepoGrabber.js');
-var scrapeService = require('./services/scraping.js');
 var queryService = require('./services/query.js');
 var fileSystemUtilities = require('./services/fileSystem/utilities');
 var retirejs = require('./services/retirejs/retire.js');
 var scanjs = require('./services/scanjs/scanner.js');
 
-var getNextGitHubRepo = function() {
-  scrapeService.scrapeUrls(function(){
-    queryService.query(function(urlList){
-      downloadService.readListOfFiles(urlList, function(parseList){
-        console.log('Initiating parsing of repos');
-        parseService.parseFile(parseList, function(){
-          console.log('Completed execution of service cycle');
-          getNextGitHubRepo();
-        });
+// async library to make sure we don't delete files before db is updated
+
+var processRepo = function(repoID) {
+  // gets individual records from the database and then returns an array with at least 3 of them
+  console.log('starting queryService to get repo...');
+  queryService.query(repoID, function(repoObj){
+    // makes dirs and downloads repos for chosen repo
+    console.log('starting downloadService to DL repos...');
+    downloadService.readListOfFiles(repoObj, function(repoID){
+      // db connection with our Repos
+      var repos = db.get('Repos');
+      var fullRelativePath = __dirname + '/../git_data/' + repoID;
+
+      // async library function that lets you run async functions in parallel and wait until 
+      // they're all finished before continuing (we wait for everything to finish before deleting files)
+      async.series([
+          function(callback){
```

Comment:
No need to address this now but pulling these inner functions out could help tame this beast.